docs/ref/contrib/gis/install/spatialite.txt

```diff
@@ -5,20 +5,30 @@ Installing Spatialite
 `SpatiaLite`__ adds spatial support to SQLite, turning it into a full-featured
 spatial database.
 
-Check first if you can install Spatialite from system packages or binaries. For
-example, on Debian-based distributions, try to install the ``spatialite-bin``
-package. For Mac OS X, follow the
-:ref:`specific instructions below<spatialite_macosx>`. For Windows, you may
-find binaries on `Gaia-SINS`__ home page. In any case, you should always
-be able to :ref:`install from source<spatialite_source>`.
+First, check if you can install Spatialite from system packages or binaries.
 
-When you are done with the installation process, skip to :ref:`create_spatialite_db`.
+For example, on Debian-based distributions, try to install the
+``spatialite-bin`` package. For distributions that package SpatiaLite 4.2+,
+install ``libsqlite3-mod-spatialite``.
+
+For Mac OS X, follow the :ref:`instructions below<spatialite_macosx>`.
+
+For Windows, you may find binaries on the `Gaia-SINS`__ home page.
+
+In any case, you should always be able to :ref:`install from source
+<spatialite_source>`.
 
 __ https://www.gaia-gis.it/fossil/libspatialite
 __ http://www.gaia-gis.it/gaia-sins/
 
 .. _spatialite_source:
 
+.. admonition:: ``SPATIALITE_LIBRARY_PATH`` setting required for SpatiaLite 4.2+
+
+    If you're using SpatiaLite 4.2+, you must put this in your settings::
+
+        SPATIALITE_LIBRARY_PATH = 'mod_spatialite'
```

Comment:
I thought of that too, however, `find_library('mod_spatialite')` returns `None`, even when it's properly installed. I'm not sure what the correct fallback to detect its presence is. We could possibly blindly set `spatialite_lib = 'mod_spatialite'`, remove the `ImproperlyConfigured` in `__init__()` and rely on a failure on `get_new_connection()`.